client/src/app/pages/create-job/create-job.component.ts

```diff
@@ -1,12 +1,86 @@
-import { Component } from '@angular/core';
+import { Component, OnInit, signal } from '@angular/core';
+import { Auth } from '@angular/fire/auth';
+import {
+  FormBuilder,
+  FormControl,
+  FormGroup,
+  ReactiveFormsModule,
+  Validators,
+} from '@angular/forms';
+import { Router } from '@angular/router';
+import { ToastrService } from 'ngx-toastr';
+import { UserStoreService } from '../../services/user-store.service';
+import { TagInputModule } from 'ngx-chips';
+import { JobService } from '../../services/job.service';
+import { JobLocationType, JobType, Skill } from '../../../../../shared/src/types/job';
+import { SkillsService } from '../../services/skills.service';
 
 @Component({
   selector: 'app-create-job',
   standalone: true,
-  imports: [],
+  imports: [ReactiveFormsModule, TagInputModule],
   templateUrl: './create-job.component.html',
   styleUrl: './create-job.component.css'
 })
-export class CreateJobComponent {
+export class CreateJobComponent implements OnInit{
+  public form!: FormGroup;
+  public allowedSkills: string[] = [];
+
+  constructor(
+    private formBuilder: FormBuilder,
+    private jobService: JobService,
+    private skillsService: SkillsService,
+    private router: Router,
+    private toastr: ToastrService,
+    private auth: Auth,
+    public userStore: UserStoreService,
+  ) {}
+
+  ngOnInit(): void {
+
+    this.skillsService.getSkills().subscribe({
+      next: (skills) => {
+        this.allowedSkills = skills.map((skill) => skill.name);
+      },
+    });
+    
+    this.form = this.formBuilder.group({
+      position: new FormControl<string>('', Validators.required),
+      type: new FormControl<JobType>(JobType.FULL_TIME, Validators.required),
+      locationType: new FormControl<JobLocationType>(JobLocationType.ON_SITE, Validators.required),
+      description: new FormControl<string>('', Validators.required),
+      skills: new FormControl<string[]>([], Validators.required)
+    });
+  }
+
+  async onSubmit(){
+
+    if (!this.form.valid) {
+      return this.toastr.error('Missing inputs.');
+    }
+
+    const formData = new FormData();
```

Comment:
You could just send a JSON request  body instead of formdata since there's no image upload.